diff -Naur nvidia-settings-410.57.old/src/libXNVCtrl/Makefile nvidia-settings-410.57/src/libXNVCtrl/Makefile
--- nvidia-settings-410.57.old/src/libXNVCtrl/Makefile	2018-09-25 21:51:03.794718036 +0200
+++ nvidia-settings-410.57/src/libXNVCtrl/Makefile	2018-09-25 21:51:14.815916043 +0200
@@ -31,7 +31,7 @@
 # assign variables
 ##############################################################################
 
-LIBXNVCTRL = libXNVCtrl.a
+LIBXNVCTRL = libXNVCtrl.so
 
 LIBXNVCTRL_PROGRAM_NAME = "libXNVCtrl"
 
@@ -58,8 +58,11 @@
 
 all: $(LIBXNVCTRL)
 
-$(LIBXNVCTRL) : $(OBJS)
-	$(AR) ru $@ $(OBJS)
+$(LIBXNVCTRL): $(OBJS)
+	$(RM) $@ $@.*
+	$(CC) -shared -Wl,-soname=$@.0 -o $@.$(NVIDIA_VERSION) $(LDFLAGS) $^ -lXext -lX11
+	ln -s $@.$(NVIDIA_VERSION) $@.0
+	ln -s $@.0 $@
 
 # define the rule to build each object file
 $(foreach src,$(SRC),$(eval $(call DEFINE_OBJECT_RULE,TARGET,$(src))))
diff -Naur nvidia-settings-410.57.old/src/Makefile nvidia-settings-410.57/src/Makefile
--- nvidia-settings-410.57.old/src/Makefile	2018-09-25 21:51:03.793718017 +0200
+++ nvidia-settings-410.57/src/Makefile	2018-09-25 21:52:06.743848992 +0200
@@ -110,9 +110,12 @@
 # assignments should be used)
 ##############################################################################
 
+LIBXNVCTRL = libXNVCtrl.so
+
 XNVCTRL_DIR             ?= libXNVCtrl
 XNVCTRL_MAKEFILE        ?= Makefile
-XNVCTRL_ARCHIVE         ?= $(XNVCTRL_DIR)/libXNVCtrl.a
+XNVCTRL_LIB             ?= $(XNVCTRL_DIR)/$(LIBXNVCTRL)
+
 XCONFIG_PARSER_DIR      ?= XF86Config-parser
 COMMON_UTILS_DIR        ?= common-utils
 COMMON_UNIX_DIR         ?= common-unix
@@ -280,11 +283,18 @@
 NVIDIA_SETTINGS_install: $(NVIDIA_SETTINGS)
 	$(MKDIR) $(BINDIR)
 	$(INSTALL) $(INSTALL_BIN_ARGS) $< $(BINDIR)/$(notdir $<)
+	$(MKDIR) $(LIBDIR)
+	$(INSTALL) $(INSTALL_LIB_ARGS) $(XNVCTRL_LIB).$(NVIDIA_VERSION) $(LIBDIR)
+	ln -s $(LIBXNVCTRL).$(NVIDIA_VERSION) $(LIBDIR)/$(LIBXNVCTRL).0
+	ln -s $(LIBXNVCTRL).$(NVIDIA_VERSION) $(LIBDIR)/$(LIBXNVCTRL)
+
+$(XNVCTRL_LIB):
+	$(MAKE) -C $(XNVCTRL_DIR)
 
 $(eval $(call DEBUG_INFO_RULES, $(NVIDIA_SETTINGS)))
-$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(XNVCTRL_ARCHIVE)
+$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(XNVCTRL_LIB)
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
-	    -rdynamic -o $@ $(OBJS) $(XNVCTRL_ARCHIVE) $(LIBS)
+	    -rdynamic -o $@ $(OBJS) $(XNVCTRL_LIB) $(LIBS)
 
 $(eval $(call DEBUG_INFO_RULES, $(GTK2LIB)))
 $(GTK2LIB).unstripped: $(GTK2_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
@@ -315,6 +325,7 @@
 	@$(MAKE) -C $(XNVCTRL_DIR) -f $(XNVCTRL_MAKEFILE)
 
 clean clobber:
+	$(MAKE) -C $(XNVCTRL_DIR) clean
 	rm -rf $(NVIDIA_SETTINGS) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d \
 		$(GTK2LIB) $(GTK3LIB) $(GTK2LIB_DIR) $(GTK3LIB_DIR) \
