Index: nvidia-settings-src/src/wayland-connector.h
===================================================================
--- nvidia-settings-src.orig/src/wayland-connector.h
+++ nvidia-settings-src/src/wayland-connector.h
@@ -29,7 +29,7 @@ typedef struct _wayland_lib
     void *(*fn_get_wayland_display)(void);
 } wayland_lib;
 
-wayland_lib wllib;
+wayland_lib __attribute__((__common__)) wllib;
 
 wayland_output_info *get_wayland_output_info(void);
 void *get_wayland_display(void);
