Index: nvidia-settings-dev_gpu_drv_cuda_a-20231002_33354363/src/libXNVCtrl/Makefile
===================================================================
--- nvidia-settings-dev_gpu_drv_cuda_a-20231002_33354363.orig/src/libXNVCtrl/Makefile
+++ nvidia-settings-dev_gpu_drv_cuda_a-20231002_33354363/src/libXNVCtrl/Makefile
@@ -50,8 +50,10 @@ LDFLAGS += $(XNVCTRL_LDFLAGS)
 
 .PHONY: all
 all: $(LIBXNVCTRL)
+all: $(LIBXNVCTRL_SHARED)
 
 .PHONY: clean
 clean:
 	rm -rf $(LIBXNVCTRL) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d
+	rm -f $(LIBXNVCTRL_SHARED) $(LIBXNVCTRL_SHARED).*
Index: nvidia-settings-dev_gpu_drv_cuda_a-20231002_33354363/src/libXNVCtrl/xnvctrl.mk
===================================================================
--- nvidia-settings-dev_gpu_drv_cuda_a-20231002_33354363.orig/src/libXNVCtrl/xnvctrl.mk
+++ nvidia-settings-dev_gpu_drv_cuda_a-20231002_33354363/src/libXNVCtrl/xnvctrl.mk
@@ -34,7 +34,8 @@ ifndef XNVCTRL_CFLAGS
   XNVCTRL_CFLAGS := $(shell $(PKG_CONFIG) --cflags x11)
 endif
 
-LIBXNVCTRL = $(OUTPUTDIR)/libXNVCtrl.a
+LIBXNVCTRL = libXNVCtrl.a
+LIBXNVCTRL_SHARED = libXNVCtrl.so
 
 LIBXNVCTRL_SRC = $(XNVCTRL_DIR)/NVCtrl.c
 
@@ -44,3 +45,9 @@ $(eval $(call DEFINE_OBJECT_RULE,TARGET,
 
 $(LIBXNVCTRL) : $(LIBXNVCTRL_OBJ)
 	$(call quiet_cmd,AR) ru $@ $(LIBXNVCTRL_OBJ)
+
+$(LIBXNVCTRL_SHARED): $(LIBXNVCTRL_OBJ)
+	$(RM) $@ $@.*
+	$(CC) -shared -Wl,-soname=$@.0 -o $@.0.0.0 $(LDFLAGS) $^ -lXext -lX11
+	ln -s $@.0.0.0 $@.0
+	ln -s $@.0 $@
