diff -Naur nvidia-settings-430.23.old/src/libXNVCtrl/xnvctrl.mk nvidia-settings-430.23/src/libXNVCtrl/xnvctrl.mk
--- nvidia-settings-430.23.old/src/libXNVCtrl/xnvctrl.mk	2019-06-02 05:02:50.000000000 -0700
+++ nvidia-settings-430.23/src/libXNVCtrl/xnvctrl.mk	2019-06-03 13:34:59.755892544 -0700
@@ -37,7 +37,7 @@
 XNVCTRL_CFLAGS ?=
 
 
-LIBXNVCTRL = $(OUTPUTDIR)/libXNVCtrl.a
+LIBXNVCTRL = libXNVCtrl.so
 
 LIBXNVCTRL_SRC = $(XNVCTRL_DIR)/NVCtrl.c
 
@@ -45,5 +45,8 @@
 
 $(eval $(call DEFINE_OBJECT_RULE,TARGET,$(LIBXNVCTRL_SRC)))
 
-$(LIBXNVCTRL) : $(LIBXNVCTRL_OBJ)
-	$(call quiet_cmd,AR) ru $@ $(LIBXNVCTRL_OBJ)
+$(LIBXNVCTRL): $(LIBXNVCTRL_OBJ)
+	$(RM) $@ $@.*
+	$(CC) -shared -Wl,-soname=$@.0 -o $@.$(NVIDIA_VERSION) $(LDFLAGS) $^ -lXext -lX11
+	ln -s libXNVCtrl.so.$(NVIDIA_VERSION) libXNVCtrl.so.0
+	ln -s libXNVCtrl.so.0 libXNVCtrl.so
diff -Naur nvidia-settings-430.23.old/src/Makefile nvidia-settings-430.23/src/Makefile 
--- nvidia-settings-430.23.old/src/Makefile	2019-06-02 05:02:48.000000000 -0700
+++ nvidia-settings-430.23/src/Makefile	2019-06-03 13:44:35.013632597 -0700
@@ -103,7 +103,10 @@
 # assignments should be used)
 ##############################################################################
 
+LIBXNVCTRL = libXNVCtrl/libXNVCtrl.so
+
 XNVCTRL_DIR             ?= libXNVCtrl
+XNVCTRL_LIB             ?= $(LIBXNVCTRL)
 XCONFIG_PARSER_DIR      ?= XF86Config-parser
 COMMON_UTILS_DIR        ?= common-utils
 COMMON_UNIX_DIR         ?= common-unix
@@ -223,6 +223,7 @@
 CFLAGS     += -I $(COMMON_UTILS_DIR)
 CFLAGS     += -I $(VIRTUAL_RESOLUTIONS_DIR)
 CFLAGS     += -I $(OUTPUTDIR)
+CFLAGS     += -I $(OUTPUTDIR)/libXNVCtrl
 CFLAGS     += $(VDPAU_CFLAGS)
 CFLAGS     += $(DBUS_CFLAGS)
 CFLAGS     += -DPROGRAM_NAME=\"nvidia-settings\"
@@ -271,11 +274,16 @@
 NVIDIA_SETTINGS_install: $(NVIDIA_SETTINGS)
 	$(MKDIR) $(BINDIR)
 	$(INSTALL) $(INSTALL_BIN_ARGS) $< $(BINDIR)/$(notdir $<)
+	$(MKDIR) $(LIBDIR)
+	$(INSTALL) $(INSTALL_LIB_ARGS) $(XNVCTRL_LIB)* $(LIBDIR)
+
+$(XNVCTRL_LIB):
+	$(MAKE) -C $(XNVCTRL_DIR)
 
 $(eval $(call DEBUG_INFO_RULES, $(NVIDIA_SETTINGS)))
-$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(LIBXNVCTRL)
+$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(XNVCTRL_LIB)
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
-	    -rdynamic -o $@ $(OBJS) $(LIBXNVCTRL) $(LIBS)
+	    -rdynamic -o $@ $(OBJS) $(XNVCTRL_LIB) $(LIBS)
 
 $(eval $(call DEBUG_INFO_RULES, $(GTK2LIB)))
 $(GTK2LIB).unstripped: $(GTK2_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
@@ -300,6 +310,7 @@
 $(foreach src,$(XCP_SRC),$(eval $(call DEFINE_OBJECT_RULE,TARGET,$(src))))
 
 clean clobber:
+	$(MAKE) -C $(XNVCTRL_DIR) clean
 	rm -rf $(NVIDIA_SETTINGS) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d \
 		$(GTK2LIB) $(GTK3LIB) $(GTK2LIB_DIR) $(GTK3LIB_DIR) \
