--- nvidia-settings-515.00.old/src/Makefile	2022-01-06 23:27:11 -0800
+++ nvidia-settings-515.00/src/Makefile	2022-01-06 23:27:11 -0800
@@ -281,7 +281,11 @@
 
 all: $(NVIDIA_SETTINGS) $(GTK2LIB) $(GTK3LIB) $(WAYLANDLIB)
 
-install: NVIDIA_SETTINGS_install NVIDIA_GTKLIB_install WAYLAND_LIB_install
+install: NVIDIA_SETTINGS_install NVIDIA_GTKLIB_install WAYLAND_LIB_install NVIDIA_XNVCTRL_install
+
+NVIDIA_XNVCTRL_install:
+	$(MKDIR) $(LIBDIR)
+	cp -a $(LIBXNVCTRL_SHARED)* $(LIBDIR)
 
 NVIDIA_GTKLIB_install: $(GTK2LIB) $(GTK3LIB)
 	$(MKDIR) $(LIBDIR)
@@ -304,26 +307,26 @@
 	$(INSTALL) $(INSTALL_BIN_ARGS) $< $(BINDIR)/$(notdir $<)
 
 $(eval $(call DEBUG_INFO_RULES, $(NVIDIA_SETTINGS)))
-$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(LIBXNVCTRL)
+$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(LIBXNVCTRL_SHARED)
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
-	    -rdynamic -o $@ $(OBJS) $(LIBXNVCTRL) $(LIBS)
+	    -rdynamic -o $@ $(OBJS) $(LIBXNVCTRL_SHARED) $(LIBS)
 
 $(eval $(call DEBUG_INFO_RULES, $(GTK2LIB)))
-$(GTK2LIB).unstripped: $(LIBXNVCTRL) $(GTK2_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
+$(GTK2LIB).unstripped: $(LIBXNVCTRL_SHARED) $(GTK2_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
 	$(call quiet_cmd,LINK) -shared $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
 	    -o $@ \
 	    -Wl,-soname -Wl,$(GTK2LIB_SONAME) \
 	    $(GTK2_OBJS) $(IMAGE_OBJS) $(XCP_OBJS) \
-	    $(LIBXNVCTRL) $(LIBS) $(GTK2_LIBS)
+	    $(LIBXNVCTRL_SHARED) $(LIBS) $(GTK2_LIBS)
 
 ifdef BUILD_GTK3LIB
 $(eval $(call DEBUG_INFO_RULES, $(GTK3LIB)))
-$(GTK3LIB).unstripped: $(LIBXNVCTRL) $(GTK3_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
+$(GTK3LIB).unstripped: $(LIBXNVCTRL_SHARED) $(GTK3_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
 	$(call quiet_cmd,LINK) -shared $(CFLAGS) $(LDFLAGS)  $(BIN_LDFLAGS) \
 	    -o $@ \
 	    -Wl,-soname -Wl,$(GTK3LIB_SONAME) \
 	    $(GTK3_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) \
-	    $(LIBXNVCTRL) $(LIBS) $(GTK3_LIBS)
+	    $(LIBXNVCTRL_SHARED) $(LIBS) $(GTK3_LIBS)
 endif
 
 ifdef BUILD_WAYLANDLIB
@@ -347,10 +347,10 @@
 	rm -rf $(NVIDIA_SETTINGS) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d \
 		$(GTK2LIB) $(GTK3LIB) $(GTK2LIB_DIR) $(GTK3LIB_DIR) \
		$(WAYLANDLIB) $(WAYLANDLIB_DIR) \
-		$(IMAGE_HEADERS) $(LIBXNVCTRL)
+		$(IMAGE_HEADERS) $(LIBXNVCTRL_SHARED)
 
 $(foreach src,$(GTK_SRC), \
     $(eval $(call DEFINE_OBJECT_RULE_WITH_DIR,TARGET,$(src),$(GTK2LIB_DIR))))
 
 ifdef BUILD_GTK3LIB
