diff -Naur nvidia-settings-430.26.old/src/libXNVCtrl/utils.mk nvidia-settings-430.26/src/libXNVCtrl/utils.mk
--- nvidia-settings-430.26.old/src/libXNVCtrl/utils.mk	2019-06-15 13:36:23.722337381 +0200
+++ nvidia-settings-430.26/src/libXNVCtrl/utils.mk	2019-06-16 08:43:55.252949232 +0200
@@ -64,7 +64,7 @@
 
 INSTALL               ?= install
 INSTALL_BIN_ARGS      ?= -m 755
-INSTALL_LIB_ARGS      ?= -m 644
+INSTALL_LIB_ARGS      ?= -m 755
 INSTALL_DOC_ARGS      ?= -m 644
 
 M4                    ?= m4
diff -Naur nvidia-settings-430.26.old/utils.mk nvidia-settings-430.26/utils.mk
--- nvidia-settings-430.26.old/utils.mk	2019-06-15 13:36:23.751337922 +0200
+++ nvidia-settings-430.26/utils.mk	2019-06-16 08:43:36.155596342 +0200
@@ -64,7 +64,7 @@
 
 INSTALL               ?= install
 INSTALL_BIN_ARGS      ?= -m 755
-INSTALL_LIB_ARGS      ?= -m 644
+INSTALL_LIB_ARGS      ?= -m 755
 INSTALL_DOC_ARGS      ?= -m 644
 
 M4                    ?= m4
