diff -Naur nvidia-settings-375.10.old/samples/Makefile nvidia-settings-375.10/samples/Makefile
--- nvidia-settings-375.10.old/samples/Makefile	2016-10-22 10:51:07.528013352 +0200
+++ nvidia-settings-375.10/samples/Makefile	2016-10-22 10:51:40.868748645 +0200
@@ -59,6 +59,7 @@
 LDFLAGS               += -L $(XNVCTRL_DIR)
 LIBS                  += -lXNVCtrl -lXext -lX11
 
+LDFLAGS               += -Wl,--as-needed
 
 ##############################################################################
 # samples
diff -Naur nvidia-settings-375.10.old/src/libXNVCtrl/utils.mk nvidia-settings-375.10/src/libXNVCtrl/utils.mk
--- nvidia-settings-375.10.old/src/libXNVCtrl/utils.mk	2016-10-22 10:51:07.519013154 +0200
+++ nvidia-settings-375.10/src/libXNVCtrl/utils.mk	2016-10-22 10:51:40.869748667 +0200
@@ -42,6 +42,8 @@
 LDFLAGS               ?=
 BIN_LDFLAGS           ?=
 
+LDFLAGS               += -Wl,--as-needed
+
 HOST_CC               ?= $(CC)
 HOST_LD               ?= $(LD)
 HOST_CFLAGS           ?= $(CFLAGS)
@@ -63,7 +65,7 @@
 
 INSTALL               ?= install
 INSTALL_BIN_ARGS      ?= -m 755
-INSTALL_LIB_ARGS      ?= -m 644
+INSTALL_LIB_ARGS      ?= -m 755
 INSTALL_DOC_ARGS      ?= -m 644
 
 M4                    ?= m4
diff -Naur nvidia-settings-375.10.old/utils.mk nvidia-settings-375.10/utils.mk
--- nvidia-settings-375.10.old/utils.mk	2016-10-22 10:51:07.517013110 +0200
+++ nvidia-settings-375.10/utils.mk	2016-10-22 10:51:40.870748689 +0200
@@ -42,6 +42,8 @@
 LDFLAGS               ?=
 BIN_LDFLAGS           ?=
 
+LDFLAGS               += -Wl,--as-needed
+
 HOST_CC               ?= $(CC)
 HOST_LD               ?= $(LD)
 HOST_CFLAGS           ?= $(CFLAGS)
@@ -63,7 +65,7 @@
 
 INSTALL               ?= install
 INSTALL_BIN_ARGS      ?= -m 755
-INSTALL_LIB_ARGS      ?= -m 644
+INSTALL_LIB_ARGS      ?= -m 755
 INSTALL_DOC_ARGS      ?= -m 644
 
 M4                    ?= m4
