diff -Naur nvidia-settings-364.12.old/samples/Makefile nvidia-settings-364.12/samples/Makefile
--- nvidia-settings-364.12.old/samples/Makefile	2016-03-22 19:38:48.851335618 +0100
+++ nvidia-settings-364.12/samples/Makefile	2016-03-22 19:40:57.790490526 +0100
@@ -59,6 +59,7 @@
 LDFLAGS               += -L $(XNVCTRL_DIR)
 LIBS                  += -lXNVCtrl -lXext -lX11
 
+LDFLAGS               += -Wl,--as-needed
 
 ##############################################################################
 # samples
diff -Naur nvidia-settings-364.12.old/src/libXNVCtrl/utils.mk nvidia-settings-364.12/src/libXNVCtrl/utils.mk
--- nvidia-settings-364.12.old/src/libXNVCtrl/utils.mk	2016-03-22 19:38:48.842335467 +0100
+++ nvidia-settings-364.12/src/libXNVCtrl/utils.mk	2016-03-22 19:40:57.791490543 +0100
@@ -42,6 +42,8 @@
 LDFLAGS               ?=
 BIN_LDFLAGS           ?=
 
+LDFLAGS               += -Wl,--as-needed
+
 HOST_CC               ?= $(CC)
 HOST_LD               ?= $(LD)
 HOST_CFLAGS           ?= $(CFLAGS)
@@ -61,7 +63,7 @@
 
 INSTALL               ?= install
 INSTALL_BIN_ARGS      ?= -m 755
-INSTALL_LIB_ARGS      ?= -m 644
+INSTALL_LIB_ARGS      ?= -m 755
 INSTALL_DOC_ARGS      ?= -m 644
 
 M4                    ?= m4
diff -Naur nvidia-settings-364.12.old/utils.mk nvidia-settings-364.12/utils.mk
--- nvidia-settings-364.12.old/utils.mk	2016-03-22 19:38:48.851335618 +0100
+++ nvidia-settings-364.12/utils.mk	2016-03-22 19:40:57.792490560 +0100
@@ -42,6 +42,8 @@
 LDFLAGS               ?=
 BIN_LDFLAGS           ?=
 
+LDFLAGS               += -Wl,--as-needed
+
 HOST_CC               ?= $(CC)
 HOST_LD               ?= $(LD)
 HOST_CFLAGS           ?= $(CFLAGS)
@@ -61,7 +63,7 @@
 
 INSTALL               ?= install
 INSTALL_BIN_ARGS      ?= -m 755
-INSTALL_LIB_ARGS      ?= -m 644
+INSTALL_LIB_ARGS      ?= -m 755
 INSTALL_DOC_ARGS      ?= -m 644
 
 M4                    ?= m4
