Index: nvidia-settings-dev_gpu_drv_cuda_a-20230124_32333255/doc/nvidia-settings.desktop
===================================================================
--- nvidia-settings-dev_gpu_drv_cuda_a-20230124_32333255.orig/doc/nvidia-settings.desktop
+++ nvidia-settings-dev_gpu_drv_cuda_a-20230124_32333255/doc/nvidia-settings.desktop
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Type=Application
-Encoding=UTF-8
 Name=NVIDIA X Server Settings
 Comment=Configure NVIDIA X Server Settings
-Exec=__UTILS_PATH__/nvidia-settings
+Exec=nvidia-settings
 Icon=nvidia-settings
-Categories=__NVIDIA_SETTINGS_DESKTOP_CATEGORIES__
+Categories=System;Settings
+Keywords=nvidia;vulkan;opengl;driver
 
 # Translation by Marcin Mikołajczak
 Name[pl]=Ustawienia serwera X NVIDIA
